components/org.wso2.config.diff.creater/src/main/java/org/wso2/config/diff/creater/ConfigDiffChecker.java

```diff
@@ -0,0 +1,339 @@
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * Main Class for migration configs.
+ */
+class ConfigDiffChecker {
+
+    private static final Log log = LogFactory.getLog(ConfigDiffChecker.class);
+
+    /**
+     * Find diff of XML and Properties files and write to output CSV files.
+     *
+     * @param configLoader    ConfigLoader which has separated file maps.
+     * @param outputGenerator OutputGenerator class.
+     * @throws IOException  IOException.
+     * @throws SAXException SAXException.
+     */
+    public void findConfigDiff(ConfigLoader configLoader, OutputGenerator outputGenerator) throws IOException,
+            SAXException {
+
+        Map<String, String> existingTags = configLoader.readFromCSV(new URL(MigrationConstants.CATALOG_URL));
+
+        checkXMLDiff(configLoader.getDefaultXMLFiles(), configLoader.getMigratedXMLFiles(),
+                configLoader.getJ2TemplateFiles(), existingTags, outputGenerator.getOutputCSV(),
+                outputGenerator.getOutputKeyValues());
+
+        checkPropertyDiff(configLoader.getDefaultPropertiesFiles(), configLoader.getMigratedPropertiesFiles(),
+                configLoader.getJ2TemplateFiles(), existingTags, outputGenerator.getOutputCSV(),
+                outputGenerator.getOutputKeyValues());
+    }
+
+    /**
+     * Check whether property files are in templates or not, if true find diff.
+     *
+     * @param defaultPropertiesFiles  default property files map.
+     * @param migratedPropertiesFiles migrated property files map.
+     * @param j2TemplateFiles         template property files .
+     * @param existingTags            map of existing knowledge.
+     * @param outputCSVFile           output csv file.
+     * @param keyValueFile            output key-value file.
+     * @throws IOException IOException
+     */
+    private void checkPropertyDiff(Map<String, File> defaultPropertiesFiles,
+                                   Map<String, File> migratedPropertiesFiles, Map<String, File> j2TemplateFiles,
+                                   Map<String, String> existingTags, File outputCSVFile, File keyValueFile)
+            throws IOException {
+
+        for (Map.Entry<String, File> entry : migratedPropertiesFiles.entrySet()) {
+
+            if (isFileTemplated(defaultPropertiesFiles, j2TemplateFiles, entry)) {
+
+                Set<Map.Entry<String, String>> changedPropertyDiffSet =
+                        findDiffPropertiesFiles(defaultPropertiesFiles.get(entry.getKey()),
+                                migratedPropertiesFiles.get(entry.getKey()));
+                writePropertiesDiffToCSV(migratedPropertiesFiles.get(entry.getKey()), existingTags,
```

Comment:
Better to define a separate variable and pass as parameters, rather than passing inline